### DISCUSSION TITLE
CLI convert command

### DISCUSSION THREAD

**Discussion by eadmaster** [Toblerity/Fiona#1323] [Ideas]
add a `convert` command in the [cli interface](https://github.com/Toblerity/Fiona/blob/master/docs/cli.rst) to convert from one dataset format to another. e.g.:

````
$ fiona convert docs/data/test_uk.shp test_uk.json
$ fiona convert docs/data/test_uk.shp test_uk.gpx
````

**Comment by sgillies** [comment]
@eadmaster at the moment, conversion with Fiona is done by piping the output of `fio cat` to the input of `fio load` like this:

```fio cat example.shp | fio load -f GPKG example.gpkg```

It's relatively slow because of the serialization and deserialization between programs. But it's extendable via other filter-like programs like `jq` or https://fio-planet.readthedocs.io/en/latest/.

A general purpose conversion command, without serialization, would be useful but complicated. One look at the options of https://gdal.org/programs/ogr2ogr.html shows how complicated such a command can get when all of the filters and transforms must be built-in.
